components/automate-elasticsearch/habitat/hooks/post-run

```diff
@@ -0,0 +1,69 @@
+#!{{pkgPathFor "core/bash"}}/bin/bash
+
+exec 2>&1
+
+source {{pkg.svc_config_path}}/health_check
+
+{{#if svc.me.cfg.http-host ~}}
+HOST="{{svc.me.sys.ip}}"
+
+{{else}}
+HOST="127.0.0.1"
+
+{{/if ~}}
+
+{{! Handlebars doesn't allow else-if style statements so we have to handle
+    two exit early cases before rendering the post hook if either of those
+    conditions are not met
+~}}
+
+{{#if cfg.disable ~}}
+# ElasticSearch has been disabled via config, exit early.
+touch ${POST_RUN_SENTINEL}
+exit 0
+
+{{/if ~}}
+
+{{#if cfg.deprecated.external_es ~}}
+# We're using the deprecated external ElasticSearch proxy node, exit early.
+touch ${POST_RUN_SENTINEL}
+exit 0
+
+{{/if ~}}
+
+# ElasticSearch >= 6 doesn't allow configuring index settings in the elasticsearch.yml.
+# We're using this hook to configure the default index settings after the service
+# has initialized enough to service queries.
+
+# The post-run hook is fired immediately after the run hook, therefore it's
+# unlikely that elasticsearch is healthy. We call the health-check wait_until_healthy
+# function which waits until ElasticSearch returns green or yellow.
+
+# We give it a default timeout of 600 seconds to ensure that the post-run
+# hook will eventually time out. We do this because Habitat does not currently
+# reap hooks if the service is unloaded. We don't want to end up in a place
```

Comment:
I think weird stuff is going to happen here if ES is in a failure loop in that these post-run hooks are going to likely run longer than it takes ES to start again and fire off another post-run hook. It'll probably be fine if it gets fixed quickly, but if it goes for a while, we're going to have a lot of health checks against ES running every second

Reply:
A better strategy might be to check the health, sleep for a few seconds if it fails, then return 1. That should get around 7494